corehq/apps/users/models.py

```diff
@@ -3041,3 +3041,26 @@ def role(self):
         elif self.domain:
             return CouchUser.from_django_user(self.user).get_domain_membership(self.domain).role
         return None
+
+
+class UserHistory(models.Model):
+    """
+    HQ Adaptation of Django's LogEntry model
+    """
+    CREATE = 1
+    UPDATE = 2
+    DELETE = 3
```

Comment:
> I don't expect the size difference is going to be an issue on this table.

You mean you don't expect this to have good number of records?
I was thinking this table would be big enough to consider this. Here are the numbers so far on production.
```
audit log entries from 2020 and 2021
create logs: 349,298
update logs: 358,434
delete logs: 4,981
```
The logging feature currently in use was released in June 2020.

We will be tracking updates through bulk upload as well now. So the growth would now be higher.

Plus, I am expecting, the display in the report would be done differently,
"Added, Updated and Deleted". 
Django also provides a way to get the string value from choice using "get_{field}_display", so that can be used elsewhere to show human friendly value.

Let me know if that does not sound reasonable.

Reply:
not a big issue, just that except for very large tables it's not usually necessary to use integers. The sizes you mention aren't that large. The places where you end up seeing the integers are in errors, tests and debugging.

Reply:
> The sizes you mention aren't that large

That is good to know. I generally sense such numbers as large. Just inexperience in working real large tables :)